### PR TITLE
revset: add working_copies() function

### PR DESCRIPTION
**Pull request by ckoehler**
It includes the working copy commit of every workspace of the repo.

Implements #3384

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by ckoehler** [comment]
Not sure how to test this, would appreciate some pointers. Is there an easy way to create a repo with two workspaces, create some commits, and then compare those to the output of `working_copies()`? Sounds involved.

**Comment by martinvonz** [comment]
> Not sure how to test this, would appreciate some pointers. Is there an easy way to create a repo with two workspaces, create some commits, and then compare those to the output of `working_copies()`? Sounds involved.

Maybe it can be done similar to how we do it for `@` here: https://github.com/martinvonz/jj/blob/7b3f8e8cb6cc33e5f183052822e081160bba564f/lib/tests/test_revset.rs#L331-L372

File: lib/src/revset.rs [hunk 1]
```diff
@@ -275,6 +275,7 @@ pub const GENERATION_RANGE_EMPTY: Range<u64> = 0..0;
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub enum RevsetCommitRef {
     WorkingCopy(WorkspaceId),
+    WorkingCopies(),
```
Comment (yuja) [inline lib/src/revset.rs:278]: nit: omit `()`

File: lib/src/revset.rs [hunk 2]
```diff
@@ -2139,6 +2148,15 @@ fn resolve_commit_ref(
                 })
             }
         }
+        RevsetCommitRef::WorkingCopies() => {
+            let wc_commits: Vec<CommitId> = repo
+                .view()
+                .wc_commit_ids()
+                .iter()
+                .map(|(_, v)| v.clone())
+                .collect();
```
Comment (yuja) [inline lib/src/revset.rs:2157]: nit: `.values().cloned()`?

File: lib/src/revset.rs (hunk 1 above)
Comment (ckoehler) [reply lib/src/revset.rs:278]: Definitely.

File: lib/src/revset.rs (hunk 2 above)
Comment (ckoehler) [reply lib/src/revset.rs:2157]: Also `collect_vec()` and remove the type hint. Thanks!

**Review by yuja — APPROVED** [review]
thanks.

**Pull request merged**